Core/src/main/java/org/tribuo/util/Util.java

```diff
@@ -517,6 +537,25 @@ public static int sampleFromCDF(double[] cdf, Random rng) {
         }
     }
 
+    /**
+     * Samples an index from the supplied cdf.
+     * @param cdf The cdf to sample from.
+     * @param rng The rng to use.
+     * @return A sample.
+     */
+    public static int sampleFromCDF(double[] cdf, RandomGenerator rng) {
+        if (Math.abs(cdf[cdf.length-1] - 1.0) > 1e-6) {
```

Comment:
Why to we have a different epsilon here vs. in validatePMF above?

Reply:
I don't remember, I'll make them uniform. I think we use 1e-6 in most places for the zero check.

Reply:
So it's that in some cases the CDF can come from floats upcast into doubles, where the tolerance is less tight, but the validate function is used for doubles. I'll add some documentation.